recommended-type-checked.js

```diff
@@ -0,0 +1,52 @@
+module.exports = {
+  extends: [
+    require.resolve('./index'),
+    'plugin:@typescript-eslint/recommended-type-checked'
+  ],
+  
+  // the ts-eslint recommended ruleset sets the parser so we need to set it back
+  parser: require.resolve('vue-eslint-parser'),
+
+  parserOptions: {
+    parser: require('typescript-eslint-parser-for-extra-files')
```

Comment:
@segevfiner Curiously, what problems were you running into that required this dependency?

Reply:
The rules that use type checking see the linked issues in the PR description.